packages/react-composites/tests/browser/common/utils.ts

```diff
@@ -402,3 +407,21 @@ const enableTooltips = async (page: Page): Promise<void> => {
 const hideVideoLoadingSpinner = async (page: Page): Promise<void> => {
   await page.addStyleTag({ content: '[data-ui-id="stream-media-loading-spinner"] {display: none}' });
 };
+
+/**
+ * Helper function for waiting for file type icons.
+ */
+const awaitFileTypeIcon = async (page: Page): Promise<void> => {
+  const fileTypeIconId: string = dataUiId(IDS.fileTypeIcon);
+  await waitForFunction(
+    page,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (args: any) => {
+      const iconNodes = Array.from(document.querySelectorAll(args.fileTypeIconId));
+      return iconNodes.every((node) => node?.querySelector('img').height > 0);
```

Comment:
I'm not totally convinced by `height > 0`, it's possible the icon exists in the DOM and has a height, but the image has not yet finished being downloaded and rendered. consider `complete` instead: https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/complete